crates/pallets/starknet/src/simulations.rs

```diff
@@ -156,54 +162,50 @@ impl<T: Config> Pallet<T> {
                     "Transaction execution failed during fee estimation: {e} {:?}",
                     std::error::Error::source(&e)
                 );
-                Err(Error::<T>::TransactionExecutionFailed)
+                Err(Error::from(e))
             }
             Ok(execution_info) => {
                 log::error!(
                     "Transaction execution reverted during fee estimation: {}",
                     // Safe due to the `match` branch order
-                    execution_info.revert_error.unwrap()
+                    &execution_info.revert_error.clone().unwrap()
                 );
-                Err(Error::<T>::TransactionExecutionFailed)
+                Err(Error::TransactionExecutionFailed(execution_info.revert_error.unwrap().to_string()))
             }
         }?;
 
         if let Some(l1_gas_usage) = tx_execution_infos.actual_resources.0.get("l1_gas_usage") {
             Ok((T::L1GasPrices::get().eth_l1_gas_price.into(), tx_execution_infos.actual_fee.0 as u128, *l1_gas_usage))
         } else {
-            Err(Error::<T>::MissingL1GasUsage.into())
+            Err(Error::MissingL1GasUsage)
         }
     }
 
     pub fn re_execute_transactions(
         transactions_before: Vec<Transaction>,
         transactions_to_trace: Vec<Transaction>,
-    ) -> Result<
-        Result<Vec<(TransactionExecutionInfo, CommitmentStateDiff)>, PlaceHolderErrorTypeForFailedStarknetExecution>,
-        DispatchError,
-    > {
+    ) -> Result<Result<Vec<(TransactionExecutionInfo, CommitmentStateDiff)>, Error>, Error> {
+        let res = Self::re_execute_transactions_inner(transactions_before, transactions_to_trace);
         storage::transactional::with_transaction(|| {
-            let res = Self::re_execute_transactions_inner(transactions_before, transactions_to_trace);
             storage::TransactionOutcome::Rollback(Result::<_, DispatchError>::Ok(Ok(res)))
         })
-        .map_err(|_| Error::<T>::FailedToCreateATransactionalStorageExecution)?
+        .map_err(|_| Error::FailedToCreateATransactionalStorageExecution)?
```

Comment:
will it be beneficial to catch the error here and pass it as an argument FailedToCreateATransactionalStorageExecution